### PR TITLE
Parsing arguments from the command line

### PR DESCRIPTION
**Pull request by mamkhari**
This PR does the following:
- Parses the Command Line Arguments with the Python Argparse module.

### CODE REVIEW

**Review by mmphego — CHANGES_REQUESTED** [review]
I do not think you fully understand how `argparse` really works. Revisit your notes and examples. Also, spend some time watching some YouTube videos for better understanding.

You are on the right path, just need some understanding.

**Comment by mamkhari** [comment]
@mmphego  Oh, okay. I thought I got is right. This means I can not present about this work, right?

**Comment by mmphego** [comment]
> @mmphego Oh, okay. I thought I got is right. This means I can not present about this work, right?

You present:
- What you have, 
- How far you went, 
- Obstacles encountered, and
- What you did to overcome them...

File: extract_unique_sensors_01.py [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
+parser = argparse.ArgumentParser(description='Extract unique sensors')
+parser.add_argument('--file_name', help='Actual name of the file')
+parser.add_argument('--url', help='Location of SensorList.txt')
+parser.add_argument('--kat_sensor', help='Name of unique sensor')
+
+args = parser.parse_args()
+
 file_name = "SensorList.txt"
 url = "http://monctl.devm.camlab.kat.ac.za/kat/doc/manuals/sensors/SensorList.txt"                                                                          
 kat_sensor = "kat.sensors"
```
Comment (mmphego) [inline extract_unique_sensors_01.py]: Why do you need to override the args you're passing in your script?

File: extract_unique_sensors_01.py [hunk 2]
```diff
@@ -1,6 +1,14 @@
+import argparse
 from urllib.request import urlopen
 from pathlib import Path
 
+parser = argparse.ArgumentParser(description='Extract unique sensors')
+parser.add_argument('--file_name', help='Actual name of the file')
+parser.add_argument('--url', help='Location of SensorList.txt')
```
Comment (mmphego) [inline extract_unique_sensors_01.py]: This can be a function!

Comment (mmphego) [inline extract_unique_sensors_01.py]: The description of the argument is a bit confusing.
From [Wikipedia](https://en.wikipedia.org/wiki/URL), 
> A Uniform Resource Locator (URL), colloquially termed a web address,[1] is a reference to a web resource that specifies its location on a computer network and a mechanism for retrieving it. A URL is a specific type of Uniform Resource Identifier (URI),[2][3] although many people use the two terms interchangeably

File: extract_unique_sensors_01.py [hunk 3]
```diff
@@ -1,6 +1,14 @@
+import argparse
 from urllib.request import urlopen
 from pathlib import Path
 
+parser = argparse.ArgumentParser(description='Extract unique sensors')
+parser.add_argument('--file_name', help='Actual name of the file')
```
Comment (mmphego) [inline extract_unique_sensors_01.py]:
```suggestion
parser.add_argument('--file_name', help='File to extract unique sensors from.')
```

File: extract_unique_sensors_01.py [hunk 4]
```diff
@@ -88,7 +96,8 @@ def splitting_extracted_sensors(candidate_sensors, sensor_name):
     return sensors_list
 
 if __name__ == "__main__":
-    sensors_data = read_sensors(file_name, url)
-    extracted_sensors = extract_all_sensors(sensors_data, kat_sensor)
-    unique_sensors_names = splitting_extracted_sensors(extracted_sensors, kat_sensor)
+    sensors_data = read_sensors(args.file_name, args.url)
+    extracted_sensors = extract_all_sensors(sensors_data, args.kat_sensor)
+    unique_sensors_names = splitting_extracted_sensors(extracted_sensors, args.kat_sensor)
+    print(f'{args.file_name}, {args.url}, {args.kat_sensor}')
```
Comment (mmphego) [inline extract_unique_sensors_01.py]: Is this for debugging?

**Review by mmphego — CHANGES_REQUESTED** [review]
You are on the right path, just need some minor changes.
:clap:

File: extract_unique_sensors_01.py (hunk 2 above)
Comment (mmphego) [reply extract_unique_sensors_01.py]: @mamkhari

File: extract_unique_sensors_01.py (hunk 1 above)
Comment (mamkhari) [reply extract_unique_sensors_01.py]: I was not aware of that, but after re-running the script I noticed that this was not needed.

Comment (mmphego) [reply extract_unique_sensors_01.py]: Please pay attention to detail, this is crucial. 
Every code commit should be reviewed by the author preferably before others review it. 

Always, think about: "What could go wrong?"

**Review by mmphego — APPROVED** [review]
Great job, this looks way better... :+1: 
Merge and update the Readme.md

**Pull request merged**